src/targets/awsLambda.ts

```diff
@@ -0,0 +1,187 @@
+  'sa-east-1',
+];
+
+const compatibleRuntimes = ['nodejs10.x', 'nodejs12.x'];
+
+/** Config options for the "aws-lambda" target. */
+interface AwsLambdaTargetOptions extends TargetConfig {
+  /** AWS access key ID, set as AWS_ACCESS_KEY_ID. */
+  awsAccessKeyId: string;
+  /** AWS secret access key, set as `AWS_SECRET_ACCESS_KEY`. */
+  awsSecretAccessKey: string;
+}
+
+/**
+ * The default layer name is used when no `AWS_LAYER_NAME`
+ * environment variable is found.
+ */
+export const defaultLayerName = 'SentryNodeServerlessSDK';
+
+/**
+ * Extracts the AWS Lambda layer name from the environment variables. If no
+ * environment variable is found, the default name is used.
+ */
+export function getAwsLayerName(): string {
+  if (!process.env.AWS_LAYER_NAME) {
+    return defaultLayerName;
+  } else {
+    return process.env.AWS_LAYER_NAME;
+  }
+}
+
+/**
+ * Target responsible for uploading files to AWS Lambda.
+ */
+export class AwsLambdaTarget extends BaseTarget {
+  /** Target name */
+  public readonly name: string = 'aws-lambda';
+  /** Target options */
+  public readonly awsLambdaConfig: AwsLambdaTargetOptions;
+  /** Name of the layer to be published */
+  static layerName: string;
+
+  public constructor(
+    config: TargetConfig,
+    artifactProvider: BaseArtifactProvider
+  ) {
+    super(config, artifactProvider);
+    this.awsLambdaConfig = this.getAwsLambdaConfig();
+    AwsLambdaTarget.layerName = getAwsLayerName();
+  }
+
+  /**
+   * Extracts AWS Lambda target options from the environment.
+   */
+  protected getAwsLambdaConfig(): AwsLambdaTargetOptions {
+    if (!process.env.AWS_ACCESS_KEY_ID || !process.env.AWS_SECRET_ACCESS_KEY) {
+      throw new ConfigurationError(
+        `Cannot perform AWS Lambda release: missing credentials.
+        Please use AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY environment variables.`
+      );
+    }
+    return {
+      awsAccessKeyId: process.env.AWS_ACCESS_KEY_ID,
+      awsSecretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+    };
+  }
+
+  /**
+   * Publishes current lambda layer zip bundle to AWS Lambda.
+   * @param _version New version to be released.
+   * @param revision Git commit SHA to be published.
+   */
+  public async publish(_version: string, revision: string): Promise<any> {
+    logger.debug('Fetching artifact list...');
+    const packageFiles = await this.getArtifactsForRevision(revision, {
+      includeNames: DEFAULT_AWS_LAMBDA_DIST_REGEX,
+    });
+
+    if (packageFiles.length == 0) {
+      reportError('Cannot release to AWS Lambda: no packages found');
+      return undefined;
+    } else if (packageFiles.length > 1) {
+      reportError(`Cannot release to AWS Lambda:
+      multiple packages with matching patterns were found.`);
+      return undefined;
+    }
+
+    const artifactBuffer = fs.readFileSync(
+      await this.artifactProvider.downloadArtifact(packageFiles[0])
+    );
+
+    const publishRegionToLambda = async (currentRegion: string) => {
```

Comment:
nit:

```suggestion
    const publishRegion = async (currentRegion: string) => {
```

For consistency: `publish` is the general method, `publishRegion` for a specific region.